lib/client.js

```diff
@@ -462,6 +474,7 @@ class Client extends EventEmitter {
     // Authenticate with client key initially
     // This is used to get $env before sending cached versions
     const creds = {
+      $req_id: crypto.randomUUID(),
```

Comment:
How about abstracting out the `$req_id` generation?

```
  $req_id: generateRequestId()
```

Eventually, we want to support passthrough of a `$req_id` from external application instead of generating a new one.  Also, it decouples id generation from the specific library (crypto).